WordPress/gutenberg
# fix: build render.php when webpack watch in progress

**PR For :- Issue https://github.com/WordPress/gutenberg/issues/49790**

## What?
- render.php isn't created in the build when created in block's src.
- See https://github.com/WordPress/gutenberg/issues/49790 for more details.

## Why?
- Based on these [findings](https://github.com/WordPress/gutenberg/issues/49790#issuecomment-1511933927) by @gziolo 
- Webpack's CopyWebpackPlugin is watching changes in the files. But it doesn't create render.php in the build folder, when the developer adds it in a src folder of a block. This PR Fixes the problem.

## How?
- renderPaths which is utilized to watch over all the render file paths, was not getting recalculated again during the watching process.
- The patch adds code to the webpack.config.js in order to recalculate the paths whenever a **/*.php file's pattern is hit in the CopyWebPack Plugin of Webpack.

```JS
let renderPaths = getRenderPropPaths();
// CopyWebPack Plugin's Pattern
{
	from: '**/*.php',
	context: getWordPressSrcDirectory(),
	noErrorOnMissing: true,
	filter: ( filepath ) => {
		renderPaths = getRenderPropPaths(); 	// Recalculate Render Pro Paths
		return (
			process.env.WP_COPY_PHP_FILES_TO_DIST ||
			renderPaths.includes( filepath )
		);
	},
},
```

## Testing Instructions
1. Create a Dynamic Block, and don't add render.php file.
2. Run the build/watch script, observe the build folder to see it reflect the src folder.
3. Add render.php to render attribute of block.json of the Block.
4. Create a render.php and see it doesn't get generated in the build folder of the Block.
5. After Patch, It will get generated, when you add the render.php to src folder.
